pkg/crypto/crypto_test.go

```diff
@@ -108,3 +109,69 @@ func decryptConfigMap(configMap string, keyPath string, noncePath string) string
 
 	return string(plaintextBytes)
 }
+
+func TestDecryptConfigMapWrongkey(t *testing.T)  {
+        
+	testString := "Sample String to Test Symmetric Encryption"
+
+        dir1, err := ioutil.TempDir("", "TestEncryption-2")
+        if err != nil {
+                t.Error(err)
+        }
+        defer os.RemoveAll(dir1)
+
+        keyPath := dir1 + "symmKeyFile"
+        wrongkeyPath := dir1 + "wrongsymmKeyFile"
+
+        key := make([]byte, 32)
+        if _, err := io.ReadFull(rand.Reader, key); err != nil {
+                t.Error(err)
+        }
+        wrongkey := make([]byte, 32)
+        if _, err := io.ReadFull(rand.Reader, wrongkey); err != nil {
+                t.Error(err)
+        }
+
+        //Encode the Key
+        keyEnc := b64.StdEncoding.EncodeToString(key)
+        //encode wrong key
+        wrongkeyEnc := b64.StdEncoding.EncodeToString(wrongkey)
+
+        err = ioutil.WriteFile(keyPath, []byte(keyEnc), 0644)
+        if err != nil {
+                t.Error(err)
+        }
+        err=ioutil.WriteFile(wrongkeyPath, []byte(wrongkeyEnc), 0644)
+	if err != nil {
+                t.Error(err)
+        }
+
+        noncePath := dir1 + "nonceFile"
+        nonce := make([]byte, 12)
+        if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+                t.Error(err)
+        }
+
+        //Encode the nonce
+        nonceEnc := b64.StdEncoding.EncodeToString(nonce)
+
+        err = ioutil.WriteFile(noncePath, []byte(nonceEnc), 0644)
+        if err != nil {
+                t.Error(err)
+        }
+
+        encConfigMap, err := EncryptConfigMap([]byte(testString), keyPath, noncePath)
+
+        if err != nil {
+                t.Errorf("Failed to encrypt configMap %v", err)
+        }
+
+        wrongdecryptedConfigMap := decryptConfigMap(encConfigMap, wrongkeyPath, noncePath)
+        fmt.Printf("What is being returned by decrypt function on getting a wrong key = %s\n", wrongdecryptedConfigMap)
```

Comment:
No need to print it here, you are going to print it anyway in case of failure below at line 173.

Reply:
Thanks for the suggestions. Will add this change as part of my next commit.